boards/v2/renesas/rzt2m_starterkit/board.yml

```diff
@@ -0,0 +1,4 @@
+board:
```

Comment:
this file now just duplicates the same information we have in `rzt2m_starter_kit.yaml` resulting in 2 yaml files used by the board which is not great.

Reply:
The fact this is used by twister now does not make it twister specific, this is a board metadata file, used now by twister, but can also be used by the build system if needed. Duplicating the same data does not make sense IMO. We can change twister to depend on and use whatever shared format we come up with and we should stick to one single board metadata file.

Reply:
> The fact this is used by twister now does not make it twister specific, this is a board metadata file,

Looking here: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/schemas/twister/platform-schema.yaml
I would say that, yes, it does contain some generic information but it also contain some very testing specific pieces, like `twister`, and `testing` sections.

Not to mention the fact that there are pieces of information duplicated here, like: `ram` and `flash`, which is not a board specific more a SoC specific information and information we have in dts (and Kconfig).

Notice how the board v2 yml can reference multiple SoCs or variants.
That doesn't fit well with the `ram` and `flash` information.

So before just merging the information (and i'm very much in favor of reducing duplicated information), then I find it important that we decide if everything should be in one file, or if dedicated files for areas makes things cleaner.
Just like we have some information in dts files, other in Kconfig.
The fact that both files are in yaml doesn't mean there should be one huge yaml file containing everything.
But we should avoid duplicating information, so let's try to discuss our options, and pros / cons.

>  Duplicating the same data does not make sense IMO. 

and there I completely agree.

Reply:
> I would say that, yes, it does contain some generic information but it also contain some very testing specific pieces, like `twister`, and `testing` sections.

and? Still this is data a board  can provide to various tooling, including build system or test runners. 

> Not to mention the fact that there are pieces of information duplicated here, like: `ram` and `flash`, which is not a board specific more a SoC specific information and information we have in dts (and Kconfig).

sure, those predate DTS and I would be happy to work on consolidating this and have 1 single source of truth, does not mean we have to leave as is and introduce a new file duplicating the data.

> So before just merging the information (and i'm very much in favor of reducing duplicated information), then I find it important that we decide if everything should be in one file, or if dedicated files for areas makes things cleaner.
> Just like we have some information in dts files, other in Kconfig.

IMO this should be 1 single file serving all tooling we have in zephyr, including build system and test runners, documentation  or whatever...

Reply:
I propose an intermediate approach, where there would be a single `board.yaml` but if the user prefers to split up the non-core info (i.e. twister, testing, etc) into a separate file they can do so by just having an `include:` statement in the `board.yaml` including the additional file with that non-core info. This means the tooling only ever looks for `board.yaml`, but users are free to split it up if they prefer.

Reply:
@carlescufi dont see much value of splitting and giving user a choice of how to include the data, after all the data need to be defined as part of the overall schema. 
This is a metadata file, having a few more keys is not going to have any impact on parsing time or build time. Those keys are not mandatory anyways.

Reply:
> sure, those predate DTS and I would be happy to work on consolidating this and have 1 single source of truth, does not mean we have to leave as is and introduce a new file duplicating the data.

so which information exactly is the new file duplicating ?
It describes the board name, which can be considered somewhat similar to the `twister` identifier, but the twister identifier field is already a duplication of today's board name obtained from the `_defconfig` file.

Any of the revision, soc, cpucluster, variant information is not existing in today's yaml file. Those are new fields.
So that leave only `vendor` to be duplication (arch field should be removed as per arch wg meeting)

> sure, those predate DTS and I would be happy to work on consolidating this and have 1 single source of truth, does not mean we have to leave as is and introduce a new file duplicating the data.

So then let's take a look at what information in existing file / twister can be consolidated and how we want to combine that information.
Today, a board like nrf5340dk has 3 twister / test yaml files:
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp.yaml
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp_ns.yaml
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/nrf5340dk_nrf5340/nrf5340dk_nrf5340_cpuapp_ns.yaml
where the differences are related to ram, flash, supported fields.

I would prefer that consolidating those files first and see what can be obtained from dts instead of describing it again here. Although my understanding of having this information duplicated was to avoid running dts, and instead being able to do twister filtering *before* deciding to run dts.

If trying to both consolidating HWMv2 with twister yaml, and on top adjusting twister to obtain more from dts, then that blows the HWMv2 PR and jeopardizes the work.

Instead I propose to work on HWMv2, get that designed and approved. Once we have that in place, then while migrating SoCs and boards, parallel work can be done on consolidating the yaml files and update twister.

Sure that will require some files to be updated after board has been migrated to HWMv2, but as the existing yaml file is mostly a rename, then there is no difference in consolidating those files during or after transitioning to HWMv2.

My proposal is:
1. Continue with HWMv2, having a dedicate file with the new fields required, as that already consolidates separate revision files, board soc / cluster / arch files spanning multiple folder, and thus reduces number of files required to describe boards with multiple SoCs / multi-core socs.
2. In parallel, start consolidating existing yaml files used by twister, so that three files are not needed to describe a single board. See which information is redundant and can be obtained from dts.
When this work is done, merge board meta data files so only a single file is needed.

> IMO this should be 1 single file serving all tooling we have in zephyr, including build system and test runners, documentation or whatever...

I would like to see the outcome first.
Wisely named files with clear content and just 10-20 lines can be easier to read and understand than a single file spanning 100 lines with information that has little in common.

So while I do agree on the principle of a single file being an advantage, I like to see the actual look, cause I have seen to many times that such know-it-all files can become messy.

So could we agree to have such work in parallel to the new hw model ?
Nothing in HWMv2 blocks extensions to the HWMv2 yaml description.

Reply:
> so which information exactly is the new file duplicating ?
> It describes the board name, which can be considered somewhat similar to the `twister` identifier, but the twister identifier field is already a duplication of today's board name obtained from the `_defconfig` file.

you will always need something to get a reference, consider this a foreign key. Right now it does duplicate vendor, arch, but my point is that we are ending up with TWO board meta data files and that does not make sense.

> If trying to both consolidating HWMv2 with twister yaml, and on top adjusting twister to obtain more from dts, then that blows the HWMv2 PR and jeopardizes the work.

I do not understand why you think this will be the case, getting data from DTS and other activities have nothing with this PR and completely orthogonal. We can attempt to define a board meta data syntax that serves everyone and you can just ignore the testing related parts in the build system etc.

We can transition step by step to the unified metadata file and I am fine with that, but we need to agree that we want to end up with 1 single board metadata file and consolidate the information in one place and avoid duplication now and in the future.

The new data related to SoCs and other things coming with this proposal are essential and needed for testing, something we desperatly needed for a long time and having all of this in one place would help with optimizing testing and improving test coverage.



> My proposal is:
> 
> 1. Continue with HWMv2, having a dedicate file with the new fields required, as that already consolidates separate revision files, board soc / cluster / arch files spanning multiple folder, and thus reduces number of files required to describe boards with multiple SoCs / multi-core socs.
> 2. In parallel, start consolidating existing yaml files used by twister, so that three files are not needed to describe a single board. See which information is redundant and can be obtained from dts.
>    When this work is done, merge board meta data files so only a single file is needed.
> 
> > IMO this should be 1 single file serving all tooling we have in zephyr, including build system and test runners, documentation or whatever...
> 
> I would like to see the outcome first. Wisely named files with clear content and just 10-20 lines can be easier to read and understand than a single file spanning 100 lines with information that has little in common.
> 
> So while I do agree on the principle of a single file being an advantage, I like to see the actual look, cause I have seen to many times that such know-it-all files can become messy.
> 
> So could we agree to have such work in parallel to the new hw model ? Nothing in HWMv2 blocks extending the HWMv2 yaml description at a later time.

yes, I am fine with that.